policies.tf

```diff
@@ -0,0 +1,48 @@
+# Task role assume policy
+data "aws_iam_policy_document" "task_assume" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+# Task logging privileges
+data "aws_iam_policy_document" "task_permissions" {
+  statement {
+    effect = "Allow"
+
+    resources = [
+      "${aws_cloudwatch_log_group.main.arn}",
+    ]
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+  }
+}
+
+# Task logging privileges
+data "aws_iam_policy_document" "task_execution_permissions" {
+  statement {
+    effect = "Allow"
+
+    resources = [
+      "*",
+    ]
+
+    actions = [
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+      "logs:CreateLogStream",
```

Comment:
The `logs:` statements here are already given (to the created log group) in the policy above this one. So you should be able delete these statements.

Reply:
Through testing we could out that this is incorrect. The task execution permission is used when the task is launched (hence it needs access to ECR) and needs access to create logs for this. Task permissions is used by the running container which means it does not need ECR permissions. So CreateLogStream is needed by both the exection role and the task role.

Reply:
AWS actually have a managed policy for this: `AmazonECSTaskExecutionRolePolicy` creating a issue for switching to this.

Reply:
Cool! So, Fargate needs the "task execution permission" to output ECS agent logs perhaps? Or some other form of logs?

Reply:
Yepp, since you don't have any instances those logs needs to go somewhere. Added an issue for using the managed policy that we will fix #2